From 49e8ec168c96e836e527d1f351d3d692b8dc6edb Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Wed, 16 Sep 2020 16:56:19 -0700
Subject: [PATCH 1605/1852] misc: xilinx-ai-engine: update reset API to use
 resource clear all function

commit 5d8e5490a2b923e921c92590ce85f4e8181ea327 from
https://github.com/Xilinx/linux-xlnx.git

Update the AI engine partition reset function to use clear all resource
function to clear the clock states and tiles in use resources.

Signed-off-by: Wendy Liang <wendy.liang@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/misc/xilinx-ai-engine/ai-engine-reset.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/xilinx-ai-engine/ai-engine-reset.c b/drivers/misc/xilinx-ai-engine/ai-engine-reset.c
index f419d2661fb5..63ec9bc095d4 100644
--- a/drivers/misc/xilinx-ai-engine/ai-engine-reset.c
+++ b/drivers/misc/xilinx-ai-engine/ai-engine-reset.c
@@ -194,9 +194,8 @@ int aie_part_reset(struct aie_partition *apart)
 	}
 
 	/* Clear tiles in use bitmap and clock state bitmap */
-	aie_resource_clear(&apart->tiles_inuse, 0, apart->tiles_inuse.total);
-	aie_resource_clear(&apart->cores_clk_state, 0,
-			   apart->cores_clk_state.total);
+	aie_resource_clear_all(&apart->tiles_inuse);
+	aie_resource_clear_all(&apart->cores_clk_state);
 
 	aie_part_set_cols_clkbuf(apart, false);
 	aie_part_set_cols_reset(apart, true);
-- 
2.31.1

